service/dap/server_test.go

```diff
@@ -397,6 +397,12 @@ func TestAttachStopOnEntry(t *testing.T) {
 		if dResp.Seq != 0 || dResp.RequestSeq != 13 {
 			t.Errorf("\ngot %#v\nwant Seq=0, RequestSeq=13", dResp)
 		}
+		// If this call to KeepAlive isn't here there's a chance that stdout will
+		// be garbage collected (since it is no longer alive long before this
+		// point), when that happens, on unix-like OSes, the read end of the pipe
+		// will be closed by the finalizer and the target process will die by
+		// SIGPIPE, which the rest of this test does not expect.
+		runtime.KeepAlive(stdout)
```

Comment:
@bcmills noticed it can be actually the `cmd` leak. defer cmd.Wait would be one way to address that cleanly, but before calling cmd.Wait, stdout pipe needs to be drained or closed. https://golang.org/pkg/os/exec/#Cmd.StdoutPipe
It's strange that this issue is surfaced this way. :-(

Reply:
There's a number of possible ways to fix this, anything that keeps a reference to the stdout pipe os.File alive will suffice. So, a goroutine that reads from scanOut, a deferred cmd.Wait, a deferred stdout.Close, passing cmd to fmt.Printf... the fact that cmd.Wait is never called doesn't really matter here, the problem is resources being released not resources being leaked.
Issues related to this get periodically reported to golang/go, the last instance I can find is https://github.com/golang/go/issues/41505 but there are much older versions.